unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Animator/Handler/AnimatorComponentHandler.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using DCL.Components;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using DCL.ECSComponents;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class AnimatorComponentHandler : IECSComponentHandler<PBAnimator>
+    {
+        private readonly DataStore_ECS7 dataStore;
+        private IDCLEntity entity;
+        private PBAnimator model;
+        internal Animation animComponent;
+        internal bool isShapeLoaded = false;
+        
+        internal Dictionary<string, AnimationClip> clipNameToClip = new Dictionary<string, AnimationClip>();
+        
+        public AnimatorComponentHandler(DataStore_ECS7 dataStoreEcs7) { dataStore = dataStoreEcs7; }
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            dataStore.animatorShapesReady.OnAdded -= AnimatorShapeReadyAdded;
+        }
+
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBAnimator model)
+        {
+            // If the entity has change, we force to check for an animatorShapeReady
+            if (this.entity != null && this.entity != entity)
+                isShapeLoaded = false;
+            
+            this.entity = entity;
+            this.model = model;
+            
+            // If shape is loaded we just update the state of the animation with the model info
+            if (isShapeLoaded)
+            {
+                UpdateAnimationState(model);
+            }
+            else
+            {
+                // If shape is not loaded, we check if it is already loaded, if not we subscribe to when it is ready
+                if (dataStore.animatorShapesReady.TryGetValue(entity.entityId, out GameObject gameObject))
+                {
+                    AnimatorShapeReadyAdded(entity.entityId, gameObject);
+                }
+                else
+                {
+                    dataStore.animatorShapesReady.OnAdded -= AnimatorShapeReadyAdded;
+                    dataStore.animatorShapesReady.OnAdded += AnimatorShapeReadyAdded;
+                }
+            }
+        }
+
+        private void AnimatorShapeReadyAdded(long entityId, GameObject gameObject)
+        {
+            if (entityId != entity.entityId)
+                return;
+          
+            dataStore.animatorShapesReady.OnAdded -= AnimatorShapeReadyAdded;
+            
+            Initialize(entity);
+            isShapeLoaded = true;
+
+            UpdateAnimationState(model);
+        }
```

Comment:
what would happen if the gltfshape component is changed after adding the animator?
i.e:
```
add shape = gltf1
add animator
modify shape = glf2
```

Reply:
Fixed